app/models/remote_execution_provider.rb

```diff
@@ -80,7 +80,11 @@ def find_ip_or_hostname(host)
 
     def find_ip(host, interfaces)
       if host_setting(host, :remote_execution_connect_by_ip)
-        interfaces.find { |i| i.ip.present? }.try(:ip)
+        if host_setting(host, :remote_execution_connect_by_ip_prefer_ipv6)
+          interfaces.find { |i| i.ip6.present? }.try(:ip6) || interfaces.find { |i| i.ip.present? }.try(:ip)
+        else
+          interfaces.find { |i| i.ip.present? }.try(:ip)
+        end
```

Comment:
```suggestion
        ip6 = interfaces.find { |i| i.ip6.present? }.try(:ip6) if host_setting(host, :remote_execution_connect_by_ip_prefer_ipv6)
        ip6 || interfaces.find { |i| i.ip.present? }.try(:ip)
```

Reply:
Thanks for your proposal, I actually did it a bit different as I suddenly realized that I overlooked something: it's a preference, meaning that the fallbacks should simply be switched. What I mean by this is, if IPv6 is preferred, it should fallback to IPv4 if no IPv6 was detected and vice versa.
Nevertheless, interesting and elegant proposal